tuf/utils/x509.go

```diff
@@ -443,6 +454,9 @@ func EncryptPrivateKey(key data.PrivateKey, role, passphrase string) ([]byte, er
 		return nil, fmt.Errorf("unable to encrypt key - invalid PEM file produced")
 	}
 	encryptedPEMBlock.Headers["role"] = role
+	if gun != "" {
```

Comment:
@riyazdf as far as Go is concerned, it'll basically be the same, but if somebody for some reason manually looks inside the PEM, it's mildly confusing to have a header with an empty value so I suggested it would be better to no include it at all if it would otherwise be `""`